packages/dd-trace/test/appsec/rasp.express.plugin.spec.js

```diff
@@ -0,0 +1,119 @@
+'use strict'
+
+const axios = require('axios')
+const agent = require('../plugins/agent')
+const getPort = require('get-port')
+const appsec = require('../../src/appsec')
+const Config = require('../../src/config')
+const path = require('path')
+const { assert } = require('chai')
+
+withVersions('express', 'express', expressVersion => {
+  describe('RASP', () => {
+    let app, server, port
+
+    before(() => {
+      return agent.load(['http'], { client: false })
+    })
+
+    before((done) => {
+      const express = require(`../../../../versions/express@${expressVersion}`).get()
+      const expressApp = express()
+
+      expressApp.get('/', (req, res) => {
+        app(req, res)
+      })
+
+      getPort().then(newPort => {
+        port = newPort
+        server = expressApp.listen(port, () => {
+          done()
+        })
+      })
+    })
+
+    beforeEach(() => {
+      appsec.enable(new Config({
+        appsec: {
+          enabled: true,
+          rules: path.join(__dirname, 'rasp_rules.json'),
+          rasp: { enabled: true }
+        }
+      }))
+    })
+
+    afterEach(() => {
+      appsec.disable()
+      app = null
+    })
+
+    after(() => {
+      server.close()
+      return agent.close({ ritmReset: false })
+    })
+
+    function getWebSpan (traces) {
+      for (const trace of traces) {
+        for (const span of trace) {
+          if (span.type === 'web') {
+            return span
+          }
+        }
+      }
+      throw new Error('web span not found')
+    }
+
+    describe('ssrf', () => {
+      ['http', 'https'].forEach(protocol => {
+        describe(`Test using ${protocol}`, () => {
+          it('Should not detect threat', async () => {
+            app = (req, res) => {
+              require(protocol).get(`${protocol}://${req.query.host}`)
+              res.end('end')
+            }
+
+            axios.get(`http://localhost:${port}/?host=www.datadoghq.com`)
+
+            await agent.use((traces) => {
+              const span = getWebSpan(traces)
+              assert.notProperty(span.meta, '_dd.appsec.json')
+            })
+          })
+
+          it('Should detect threat doing a GET request', async () => {
```

Comment:
i don't see the difference between the test with threat and without ? :thinking: is the datadog domain just whitelisted ?

Reply:
Yes, the difference is the domain.
Is not that datadog domain is whitelisted, is that ifconfig.pro is considered as a threat by the waf (I saw the example in the RFC and used it)